docs/sdk/src/reference_docs/runtime_vs_smart_contract.rs

```diff
@@ -1,6 +1,209 @@
-//! Runtime vs. Smart Contracts
+//! # Runtime vs. Smart Contracts
 //!
-//! Notes:
+//! *TL;DR*: If you need to create a *Blockchain*, then write a runtime. If you need to create a
+//! *DApp*, then write a Smart Contract.
 //!
-//! Why one can be weighed, and one MUST be metered.
-//! https://forum.polkadot.network/t/where-contracts-fail-and-runtimes-chains-are-needed/4464/3
+//! This is a comparative analysis of Substrate-based Runtimes and Smart Contracts, highlighting
+//! their main differences. Our aim is to equip you with a clear understanding of how these two
+//! methods of deploying on-chain logic diverge in their design, usage, and implications.
+//!
+//! Both Runtimes and Smart Contracts serve distinct purposes. Runtimes offer deep customization for
+//! blockchain development, while Smart Contracts provide a more accessible approach for
+//! decentralized applications. Understanding their differences is crucial in choosing the right
+//! approach for a specific solution.
+//!
+//! ## Substrate
+//! Substrate is a modular framework that enables the creation of purpose-specific blockchains. In
+//! the Polkadot ecosystem you can find two distinct approaches for on-chain code execution:
+//! [Runtime Development](#runtime-in-substrate) and [Smart Contracts](#smart-contracts).
+//!
+//! #### Smart Contracts in Substrate
+//! Smart Contracts are autonomous, programmable constructs deployed on the blockchain.
+//! In [FRAME](frame), Smart Contracts infrastructure is implemented by the
+//! [`pallet_contracts`](../../../pallet_contracts/index.html) for WASM-based contracts or the
+//! [`pallet_evm`](../../../pallet_evm/index.html) for EVM-compatible contracts. These pallets
+//! enable Smart Contract developers to build applications and systems on top of a Substrate-based
+//! blockchain.
```

Comment:
Suggestion: a diagram here that shows a substrate based chain, hosting `pallet-evm`, which itself is hosting a number of Solidity contracts.